extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultTransitSecretEngine.java

```diff
@@ -0,0 +1,90 @@
+import io.quarkus.vault.transit.VerificationResult;
+import io.quarkus.vault.transit.Verifyable;
+
+public interface VaultTransitSecretEngine {
+
+    // encrypt
+
+    /**
+     * Encrypt a regular string with a vault key configured in the transit secret engine.
+     * equivalent to:
+     * {@code encrypt(keyName, plaintext.getBytes("UTF-8"), Optional.empty());}
+     * <p>
+     * this method is usually used in conjunction with {@link #decryptString(String, String)}
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param plaintext the string to encrypt
+     * @return cipher text
+     */
+    String encryptString(String keyName, String plaintext);
+
+    /**
+     * Encrypt a regular string with a vault key configured in the transit secret engine.
+     * if the key does not exist, and the policy specifies a create capability the key will be lazily created
+     * (i.e. upsert). the key can be further customized by specifying transit encryption-key configuration
+     * properties. as vault encrypts bytes, and not strings, the plaintext string is converted to bytes using
+     * the UTF-8 charset, before being passed to the hashicorp vault encryption service.
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param plaintext the string to encrypt
+     * @param transitContext an optional context for key derivation
+     * @return cipher text
+     */
+    Ciphertext encrypt(String keyName, Plaintext plaintext, TransitContext transitContext);
+
+    /**
+     * Encrypt a list of elements. This will return a list of results, each being either an ciphertext or an error.
+     * Each element can be specify an optional context, allowing key derivation.
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param requests the list of elements to encrypt
+     * @return a list of result objects containing either a ciphertext or an error
+     */
+    List<EncryptionResult> encrypt(String keyName, List<EncryptionRequest> requests);
+
+    // decrypt
+
+    String decryptString(String keyName, String ciphertext);
+
+    Plaintext decrypt(String keyName, Ciphertext ciphertext, TransitContext transitContext);
+
+    List<DecryptionResult> decrypt(String keyName, List<DecryptionRequest> requests);
+
+    // rewrap
+
+    Ciphertext rewrap(String keyName, Ciphertext ciphertext, TransitContext transitContext);
+
+    List<EncryptionResult> rewrap(String keyName, List<RewrappingRequest> requests);
+
+    // sign
+
+    Signature sign(String keyName, Plaintext input, TransitContext transitContext);
```

Comment:
@vsevel we should also have `signString` IMHO

Reply:
I have added `Signature signString(String keyName, String input)`.
I was not convinced it was adding a lot of value. Up to discussion.

Do you think it should return `String` instead of `Signature`?

On the same idea I replaced the return type to `EncryptedData` in `EncryptedData encryptString(String keyName, String clearData)`. Do you think we should keep `String`? Now that we have the wrapper types, it feels more confusing to return a string when it is actually a cipher text.

And to be consistent I replaced the argument type in `String decryptString(String keyName, EncryptedData encryptedData)`.

What do you think?

Reply:
@vsevel But why to use the wrapper types ? It adds to the API expressiveness, but adds an extra indirection in the code in all the cases:
```
return encryptString("mykey", "hello").getValue()
vs
return encryptString("mykey", "hello")
```

By the way. what is your opinion about the overloads, ex, just have `encrypt(String key, String cleardata)` ?

Reply:
> But why to use the wrapper types ? It adds to the API expressiveness, but adds an extra indirection in the code in all the cases

true. it felt like expressiveness was more important than the added indirection. but that was my question. I will change it back.

> By the way. what is your opinion about the overloads

very good point. I have gone back and forth. the problem I had was not on `encrypt` but on `decrypt`. If I did `String encrypt(String keyName, String clearData)` then I would probably want to do `String decrypt(String keyName, EncryptedData encryptedData)` as well. But this one is more problematic because you have no idea that `encryptedData` was originally encrypted from a `String` and not just an array of `bytes`.  But may be I am thinking too much on that one. I will use the overload everywhere.